### ISSUE TITLE
adding root level pages to the searchindex

### ISSUE THREAD

**Issue by allotmentandy** [tighten/jigsaw#715]
hi, i am looking to add a number of pages at the root level of a jigsaw site to the searchIndex

there isnt a collection called pages (as there is for posts)

is there an easy way to do this? i would prefer to have a collection in the config.php file which states which php and md pages to add to the searchIndex if possible. 

hope you can help and point me in the right direction.

thanks in advance

**Comment by husseinalhammad** [comment]
You can get all generated pages in an event listener and do whatever you want with them (e.g. generate a sitemap):
https://jigsaw.tighten.com/docs/event-listeners/

Is this what you're after?

**Comment by allotmentandy** [comment]
no, i want to add a number of pages to the searchIndex. 

so i would like to add index.blade.php, gardening.blade.php and itsupport.blade.php to the searchIndex.json that is created at the end of the build process

**Comment by husseinalhammad** [comment]
How are you generating the `searchIndex.json`? Jigsaw doesn't create this by default.

Are you by any chance using one of the [starter templates](https://jigsaw.tighten.com/docs/starter-templates/)?

**Comment by allotmentandy** [comment]
yes i am using the blog template with the fuse search. you can see it in action here, https://allotmentandy.github.io/ 

thanks for your help with this issue

**Comment by husseinalhammad** [comment]
The blog starter template generates the JSON file in an event listener.

You can modify the `./listeners/GenerateIndex.php` class to include the pages you need. By default this only includes items from the posts collection:
https://github.com/tighten/jigsaw-blog-template/blob/main/listeners/GenerateIndex.php

**Comment by allotmentandy** [comment]
i tried that, but there is no collection for pages, so i wonder what i would do to create an additional collection one in the config.php and then add them in that GenerateIndex.php file. 

would it just be an array/collection of filenames?

**Comment by husseinalhammad** [comment]
Check the helper methods in the docs. You can use `getPages()`
https://jigsaw.tighten.com/docs/event-listeners/#helper-methods-in-$jigsaw
